src/flexible_type/flexible_type_detail.hpp

```diff
@@ -535,7 +535,16 @@ struct get_int_visitor {
     return dt.posix_timestamp();
   }
   inline FLEX_ALWAYS_INLINE_FLATTEN flex_int operator()(flex_float i) const {return i; }
-  inline FLEX_ALWAYS_INLINE_FLATTEN flex_int operator()(const flex_string& t) const { return std::atoll(t.c_str()); }
+  inline FLEX_ALWAYS_INLINE_FLATTEN flex_int operator()(const flex_string& t) const { 
+    char *end;
+    long long int converted = std::strtoll(t.c_str(), &end, 10); 
+    if (*end != '\0') {
```

Comment:
I rather this not use (*end) == '\0' as a comparison. The null character could occur anywhere.
Use string.data(), and check for position of the pointer instead of comparing the value at the pointer.